### ISSUE TITLE
Prerequisites not listed

### ISSUE THREAD

**Issue by exscape** [indirect/dash-rust#5]
Unfortunately, running `rake` was most certainly not the only step required. In fact, for the time being, I gave up trying to get this to work (on OS X 10.9.2), as several of the steps caused issues; most notably, installing the dependencies for bundler (which appears to be a prerequisite for dash-rust) fails for me:

$ git clone https://github.com/indirect/dash-rust.git
$ rake
rake aborted!
cannot load such file -- bundler/setup
/Users/serenity/Programming/rust_src/dash-rust/Rakefile:2:in `&lt;top (required)&gt;'
(See full trace by running task with --trace)

$ sudo gem install bundler
Fetching: bundler-1.6.2.gem (100%)
Successfully installed bundler-1.6.2
Parsing documentation for bundler-1.6.2
Installing ri documentation for bundler-1.6.2
1 gem installed

$ rake
Could not find coderay-1.1.0 in any of the sources
Run `bundle install` to install missing gems.

$ bundle install
Fetching gem metadata from https://rubygems.org/.........
Fetching additional metadata from https://rubygems.org/..
Installing coderay 1.1.0
Installing execjs 2.0.2
Installing method_source 0.8.2
Installing mini_portile 0.5.2

Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native extension.
...
checking for iconv_open() in -liconv... yes
checking for xmlParseDoc() in -lxml2... no
-------
libxml2 is missing.  please visit http://nokogiri.org/tutorials/installing_nokogiri.html for help with installing dependencies.
-----

I then tried to solve this using the answer at [this StackOverflow link](http://stackoverflow.com/questions/19643153/error-to-install-nokogiri-on-osx-10-9-maverick), but that doesn't work, either.

I fully realize that none of this is because of dash-rust, of course, but at the very least, the prerequisites should be listed.
Is bundler the only one (with its many dependencies of its own)?

**Comment by indirect** [comment]
I don't know what kind of machine you're running this on, but Dash only runs on Macs. OS X ships with both libxml2 and bundler already installed. I added a note about needing to run `bundle install`, though, it's a good point.

**Comment by exscape** [comment]
Macbook pro 15.4" Early 2011, Mavericks 10.9.2, as noted. Most certainly a standard Mac, in other words.

**Comment by indirect** [comment]
Whoops, I missed that. Thanks for letting me know, and that's reeeeeeeally weird. The version of Nokogiri used here even includes the source to libxml, and will build it if it is missing. :(

If this command did not work:

```
gem install nokogiri -- --with-xml2-include=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/include/libxml2
```

Then you probably don't have the required developer tools installed. Building the docset yourself requires not only that you have XCode and the command-line developer tools installed, it also requires that you be able to compile all of Rust from the source code, to generate the docs that will be used in the docset.

**Comment by exscape** [comment]
I have XCode 5.1.1 (5B1008) installed, and I have built Rust from source many times (I build/test it myself from the master branch), so I should have those, too! (I have run `xcode-select --install` prior.)

$ gem install nokogiri -- --with-xml2-include=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/include/libxml2
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.

With sudo in front, it fails during configure due to
checking for xmlParseDoc() in -lxml2... no
[Full command output](http://pastebin.com/MyvMJvSJ)
mkmf.log doesn't appear to show much useful information, except that it does pass "-L/Library/Ruby/Gems/2.0.0/gems/nokogiri-1.6.1/ports/x86_64-apple-darwin13.1.0/libxml2/2.8.0/lib -I/Library/Ruby/Gems/2.0.0/gems/nokogiri-1.6.1/ports/x86_64-apple-darwin13.1.0/libxml2/2.8.0/include/libxml2" and "-lxml2" to clang while building the test that fails.

The include path supplied (in the gem install command) does exist and work, though; there are 47 header files in .../libxml2/libxml in the path. This seems fairly strange; it indeed specifies that it runs configure/make/install on libxml2 during the gem install process, but it still fails to find libxml2.

The [official instructions](http://nokogiri.org/tutorials/installing_nokogiri.html) using homebrew also don't work (after adjusting for version numbers), with or without custom libiconv; same error (though with the custom libiconv, that's what it can't find, instead).

BTW, sorry for posting all of this here.

**Comment by indirect** [comment]
Ugh. That's extremely strange. I'm out of ideas, now, sorry. :( Maybe they can help you over at https://github.com/sparklemotion/nokogiri/issues ?

**Comment by exscape** [comment]
Well, I finally solved this part, at least! In case someone has the same issue:
I found [this rather old](http://chrisroos.co.uk/blog/2010-03-26-libxml2-problems-when-installing-nokogiri) guide. I used solution 1, with a small change:

```
ARCHFLAGS="-arch x86_64" gem install nokogiri
```

... as root (since root owns /Library/Ruby/Gems), which finally worked, just like that.
I did not try i386 as suggested, since I figured that is likely outdated by now; I use 64-bit Rust, 64-bit compilers and so likely 64-bit libxml2 and nokogiri, too.

BTW (not sure if I should open a separate issue): if you enter the path to the full Rust source instead of e.g. rust/doc, dash-rust will copy the entire source tree, and then fail to build the docs later. It would probably be good to check the path before it starts working. With the proper doc path, though, it works great now!
